src/libsync/theme.cpp

```diff
@@ -280,6 +280,85 @@ QIcon Theme::uiThemeIcon(const QString &iconName, bool uiHasDarkBg) const
     return QIcon(QPixmap(iconPath));
 }
 
+QPixmap Theme::pixmapForBackground(const QString &fileName, const QColor &backgroundColor) const
+{
+    Q_ASSERT(!fileName.isEmpty());
+
+    const auto iconBasePath = QStringLiteral(":/client/theme");
+
+    // some icons are present in white or black only, so, we need to check both when needed
+    const auto iconBaseColors = QStringList({ QStringLiteral("black"), QStringLiteral("white") });
+
+    const QString pixmapColor = backgroundColor.isValid() && !Theme::isDarkColor(backgroundColor) ? "black" : "white";
+
+    const QString cacheKey = fileName + QLatin1Char(',') + pixmapColor;
+
+    auto &cachedPixmap = _pixmapCache[cacheKey];
+
+    if (cachedPixmap.isNull()) {
+        if (iconBaseColors.contains(pixmapColor)) {
+            return cachedPixmap = QPixmap::fromImage(QImage(iconBasePath + QLatin1Char('/') + pixmapColor + QLatin1Char('/') + fileName));
+        }
+
+        const auto drawSvgWithCustomFillColor = [](const QString &sourceSvgPath, const QString &fillColor) {
```

Comment:
Not sure why this is not a static function inside the cpp file :)

Reply:
@FlexW Same reason as for another one. It is just used inside this single method and that's it. As soon as it becomes needed elsewhere, then, I would vouch for a static function.